src/user/AddressBookController.ts

```diff
@@ -78,8 +77,21 @@ export class AddressBookController extends BaseController<BaseConfig, AddressBoo
 		if (!isValidAddress(address)) {
 			return false;
 		}
-		this.addressBook.set(address, { address, name });
-		this.update({ addressBook: Array.from(this.addressBook.values()) });
+		const oldAddresses: any = {};
+		for (const entry in this.state.addressBook) {
+			const oldEntry = {
+				address: this.state.addressBook[entry].address,
+				name: this.state.addressBook[entry].name
+			};
+			oldAddresses[entry] = oldEntry;
+		}
+
+		const addressEntry = { address, name };
+		const newEntry: any = {};
+		newEntry[address] = addressEntry;
+
+		const combined = extend(oldAddresses, newEntry);
+		this.update({ addressBook: combined });
```

Comment:
Is there a need to rebuild dirty state like this? It looks like this entire function could just be:

```ts
set(address: string, name: string) {
	if (!isValidAddress(address)) {
		return false;
	}
	this.update({ addressBook: extend(this.state.addressBook, {
		[address]: { address, name }
	});
}
```